pulsar-client-cpp/python/pulsar_test.py

```diff
@@ -793,6 +793,43 @@ def test_get_topics_partitions(self):
                          [topic_non_partitioned])
         client.close()
 
+    def test_token_auth(self):
+        with open('/tmp/pulsar-test-data/certs/token.txt') as tf:
+            token = tf.read().strip()
+        client = Client(self.adminUrl,
+                        authentication=AuthenticationToken(token))
+
+        consumer = client.subscribe('persistent://private/auth/my-python-topic-token-auth',
+                                    'my-sub',
+                                    consumer_type=ConsumerType.Shared)
+        producer = client.create_producer('persistent://private/auth/my-python-topic-token-auth')
+        producer.send(b'hello')
+
+        msg = consumer.receive(1000)
+        self.assertTrue(msg)
+        self.assertEqual(msg.data(), b'hello')
+        client.close()
+
+    def test_token_auth_supplier(self):
+        def read_token():
+            with open('/tmp/pulsar-test-data/certs/token.txt') as tf:
+                return tf.read().strip()
+
+        client = Client(self.serviceUrl,
```

Comment:
why serviceUrl and http above? If you use http here it will still exercise both endpoints